File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for(int i = 0; i < n; i++){
            answer++;
            while(answer % 3 == 0 || String.valueOf(answer).indexOf("3") != -1) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;

    for(int i = 1; i <= n; i++){
        answer = check(answer) + 1;
    }
    
    answer -= 1;
    
    return answer;
}

int check(int a){
    while(a % 3 == 0 || a % 10 == 3 || (a / 10) % 10 == 3){
        a++;
    }
    return a;
}